File: api/DTOs/Finance/VoucherToAddNewDto.cs
using DocumentFormat.OpenXml.Office2010.Excel;

namespace api.DTOs.Finance
{
    public class VoucherToAddNewDto
    {
        public int DebitCOAId { get; set; }
        public string DebitAccountName { get; set; }
        public Id CreditCOAId { get; set; }
        public string CreditAccountName { get; set; }
        public long Amount { get; set; }
        public DateOnly VoucherDate { get; set; }
        public string Narration { get; set; }
        public bool DrEntryRequiresApproval { get; set; }
    }
}

File: client/src/app/_dtos/finance/voucherToAddNewDto.ts
export interface IVoucherToAddNewDto
{
	debitCOAId: number;
	debitAccountName: string;
	creditCOAId: number;
	creditAccountName: string;
	amount: number;
	voucherDate: Date;
	narration: string;
	drEntryRequiresApproval: boolean;
}

export class VoucherToAddNewDto implements IVoucherToAddNewDto
{
	debitCOAId = 0;
	debitAccountName = '';
	creditCOAId = 0;
	creditAccountName = '';
	amount= 0;
	voucherDate= new Date('1900-01-01');
	narration = '';
	drEntryRequiresApproval = false;
}